timdex_dataset_api/dataset.py

```diff
@@ -399,8 +427,14 @@ def write(
             use_threads=use_threads,
         )
 
-        self.log_write_statistics(start_time)
-        return self._written_files  # type: ignore[return-value]
+        # write metadata append deltas
+        if write_append_deltas:
+            for written_file in written_files:
+                self.metadata.write_append_delta_duckdb(written_file.path)  # type: ignore[attr-defined]
```

Comment:
This is the most important change for `TIMDEXDataset`, where it farms out the work of writing append deltas to an _already attached_ `TIMDEXDatasetMetadata` instance at `self.metadata`.